src/devtools/useGotoAtomsSnapshot.ts

```diff
@@ -14,27 +14,35 @@ type AtomsSnapshot = Readonly<{
 
 export function useGotoAtomsSnapshot(scope?: Scope) {
   const ScopeContext = getScopeContext(scope)
-  const scopeContainer = useContext(ScopeContext)
-  const store = scopeContainer.s
+  const { s: store, w: versionedWrite } = useContext(ScopeContext)
 
   if (!store[DEV_SUBSCRIBE_STATE]) {
     throw new Error('useGotoAtomsSnapshot can only be used in dev mode.')
   }
 
   return useCallback(
     (values: Iterable<readonly [Atom<unknown>, unknown]> | AtomsSnapshot) => {
+      const writeStore = (value: any) => {
+        if (versionedWrite) {
+          versionedWrite((version) => {
+            store[RESTORE_ATOMS](value, version)
+          })
+        } else {
+          store[RESTORE_ATOMS](value)
+        }
+      }
```

Comment:
```suggestion
      const restoreAtoms = () => {
        if (versionedWrite) {
          versionedWrite((version) => {
            store[RESTORE_ATOMS](values, version)
          })
        } else {
          store[RESTORE_ATOMS](values)
        }
      }
```

Should be good without `any`?

Reply:
Oh, I missed it. Then, I'd do this.

```ts
  return useCallback(
    (snapshot: AtomsSnapshot | Iterable<readonly [Atom<unknown>, unknown]>) => {
      const restoreAtoms = (values: Iterable<readonly [Atom<unknown>, unknown]>) => {
        if (versionedWrite) {
          versionedWrite((version) => {
            store[RESTORE_ATOMS](values, version)
          })
        } else {
          store[RESTORE_ATOMS](values)
        }
      }
      if (isIterable(snapshot)) {
        if (__DEV__) {
          console.warn(
            'snapshot as iterable is deprecated. use an object instead.'
          )
        }
        restoreAtoms(snapshot)
        return
      }
      restoreAtoms(snapshot.values)
    },
    [store, versionedWrite]
  )
```